example/server.rb

```diff
@@ -50,7 +50,11 @@
   conn = HTTP2::Server.new
   conn.on(:frame) do |bytes|
     # puts "Writing bytes: #{bytes.unpack("H*").first}"
-    sock.write bytes
+    if sock.is_a?(TCPSocket)
+      sock.send bytes, 0
```

Comment:
That's the whole issue. `IO#write` works differently from `TCPSocket#sendmsg`. I haven't figured out what the difference is.

Here is the hierarchy: `IO < BasicSocket < IPSocket < TCPSocket`.

In contrast, `OpenSSL::SSL::SSLSocket` does not inherit from any of the above, and the #write method works differently.

Yes, we need to check the class because `#sendmsg` does not work with SSLSocket.

I think we should create a Server model from the server.rb file, and then a method would encapsulate the `#write` and `#sendmsg` methods.